src/StakeManagerProxy.sol

```diff
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.26;
+
+import { ERC1967Proxy } from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+
+contract StakeManagerProxy is ERC1967Proxy {
```

Comment:
We need a contract that exposes the implementation address of the ERC1967 proxy via a public function.
Otherwise `StakeVault` is unable to determine the implementation address of the `StakeManager` as there's no way to get the value of a storage slot of an external contract within the EVM.

That proxy implementation here is the one that will be deployed for every stakemanager.